File: solutions/python/algorithm/priorityqueue/Exercise2.py
from structure.dynamic.DynamicPriorityQueueADT import DynamicPriorityQueueADT


def eliminar_por_prioridad(cola, prioridad_objetivo):
    aux = DynamicPriorityQueueADT()

    while not cola.is_empty():
        valor = cola.get_element()
        prioridad = cola.get_priority()
        cola.remove()
        if prioridad != prioridad_objetivo:
            aux.add(valor, prioridad)

    while not aux.is_empty():
        cola.add(aux.get_element(), aux.get_priority())
        aux.remove()

# Ejemplo de uso
cola = DynamicPriorityQueueADT()
cola.add(10, 2)
cola.add(20, 3)
cola.add(30, 2)
cola.add(40, 1)

eliminar_por_prioridad(cola, 2)

while not cola.is_empty():
    print(f"Valor: {cola.get_element()}, Prioridad: {cola.get_priority()}")
    cola.remove()

File: solutions/javascript/src/algorithm/priorityqueue/Exercise2.js
import DynamicPriorityQueueADT from "../../structure/dynamic/DynamicPriorityQueueADT.js";

function eliminarPorPrioridad(cola, prioridadObjetivo) {
  const aux = new DynamicPriorityQueueADT();

  while (!cola.isEmpty()) {
    const value = cola.getElement();
    const priority = cola.getPriority();

    cola.remove();
    if (priority !== prioridadObjetivo) {
      aux.add(value, priority);
    }
  }

  while (!aux.isEmpty()) {
    const value = aux.getElement();
    const priority = aux.getPriority();
    aux.remove();
    cola.add(value, priority);
  }
}

// Ejemplo
const cola = new DynamicPriorityQueueADT();
cola.add(10, 2);
cola.add(20, 3);
cola.add(30, 2);
cola.add(40, 1);

eliminarPorPrioridad(cola, 2); // Elimina 10 y 30

while (!cola.isEmpty()) {
    const value = cola.getElement();
    const priority = cola.getPriority();

  console.log(`Valor: ${value}, Prioridad: ${priority}`);
  cola.remove();
}